backend/src/test/java/com/github/boltzmann/biokiste/backend/controller/ItemControllerTest.java

```diff
@@ -27,6 +29,36 @@ void getAllItemsTest_whenNoItemsInRepo_getEmptyList(){
         Assertions.assertEquals(List.of(), actual);
     }
 
+    @Test
+    void createNewItemTest_whenNewItemInBlankRepoPosted_getNewItemAndItemIsInRepo(){
+        String jwt = createUserInLoginRepoAndGetTokenForHer();
+        ItemDto dto = ItemDto.builder().name(one().getName()).build();
+        Item actual = createNewItem(jwt, dto);
+        Assertions.assertEquals(one().getName(), actual.getName());
+    }
+
+    @Test
+    void createNewItemTest_whenOtherItemInBlankRepoPosted_getNewItemAndItemIsInRepo(){
+        String jwt = createUserInLoginRepoAndGetTokenForHer();
+        ItemDto dto = ItemDto.builder().name(two().getName()).build();
+        Item actual = createNewItem(jwt, dto);
+        Assertions.assertEquals(two().getName(), actual.getName());
+    }
+
+    @Test
+    void createNewItemTest_whenItemAlreadyInRepoPosted_getErrorItemNotAdded(){
+        String jwt = createUserInLoginRepoAndGetTokenForHer();
+        ItemDto dto = ItemDto.builder().name(two().getName()).build();
+        itemRepository.insert(Item.builder().id("1").name("two").build());
+        // When
+        webTestClient.post()
+                .uri("/api/item")
+                .headers(http -> http.setBearerAuth(jwt))
+                .bodyValue(dto)
+                .exchange()
+                .expectStatus().is4xxClientError();
+    }
+
```

Comment:
Macht es Sinn, wenn du den jwt in eine Before_Each Funktion auslagerst, um doppelten Code zu vermeiden?

Reply:
Es geht so und .. es wird jetzt einfach so gemacht!